libbeat/monitoring/inputmon/httphandler_test.go

```diff
@@ -0,0 +1,121 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package inputmon
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/elastic-agent-libs/monitoring"
+)
+
+type TestCase struct {
+	request string
+	status  int
+	method  string
+	body    string
+}
+
+var testCases = []TestCase{
+	{request: "/inputs/", status: http.StatusOK, body: `[{"gauge":13344,"id":"123abc","input":"foo"}]`},
+	{request: "/inputs", status: http.StatusOK},
+	{request: "/inputs/", method: "POST", status: http.StatusMethodNotAllowed},
+	{request: "/inputs/?XX", status: http.StatusBadRequest},
+	{request: "/inputs/?pretty", status: http.StatusOK},
+	{request: "/inputs/?type", status: http.StatusBadRequest},
+	{request: "/inputs/?type=udp", status: http.StatusOK, body: `[]`},
+	{request: "/inputs/XX", status: http.StatusNotFound},
+}
+
+func TestHandler(t *testing.T) {
+	parent := monitoring.NewRegistry()
+	reg, _ := NewInputRegistry("foo", "123abc", parent)
+	monitoring.NewInt(reg, "gauge").Set(13344)
+
+	r := mux.NewRouter()
+	s := httptest.NewServer(r)
+	defer s.Close()
+
+	if err := attachHandler(r, parent); err != nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("http://%s", s.Listener.Addr().String())
+
+	for _, tc := range testCases {
+		tc := tc
+		if tc.method == "" {
+			tc.method = http.MethodGet
+		}
+
+		t.Run(tc.method+" "+strings.ReplaceAll(tc.request, "/", "_"), func(t *testing.T) {
+			req, err := http.NewRequestWithContext(context.TODO(), tc.method, s.URL+tc.request, nil)
```

Comment:
Semantics nitpick, it should be `context.Background()`
```suggestion
			req, err := http.NewRequestWithContext(context.Background(), tc.method, s.URL+tc.request, nil)
```

Reply:
Fixed. 1bc1ecc2